### PR TITLE
Detect go binary packages

### PR DESCRIPTION
**Pull request by pombredanne**
This PR add support to scan Go binaries for their embedded packages. This is the ScanCode side.

The Go inspector PR is at
- https://github.com/aboutcode-org/go-inspector/pull/15 and  need to be merged first and released for things to work

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [x] Updated documentation pages (if applicable)
* [x] Updated CHANGELOG.rst (if applicable)

### CODE REVIEW

**Comment by pombredanne** [comment]
@AyanSinhaMahapatra ready for your review

File: tests/packagedcode/test_plugin_package.py [hunk 1]
```diff
@@ -220,9 +221,12 @@ def test_package_command_scan_winmd(self):
...
 
     def test_package_list_command(self, regen=REGEN_TEST_FIXTURES):
-        expected_file = self.get_test_loc('plugin/help.txt')
+        if on_linux:
+            expected_file = self.get_test_loc('plugin/plugins_list_linux.txt')
+        else:
+            expected_file = self.get_test_loc('plugin/plugins_list.txt')
         result = run_scan_click(['--list-packages'])
-        if regen:
+        if True:
```
Comment (AyanSinhaMahapatra) [inline tests/packagedcode/test_plugin_package.py]:
```suggestion
        if regen:
```

**Review by AyanSinhaMahapatra — APPROVED** [review]
LGTM! Thanks @pombredanne 
Can be merged with a minor nit resolved.

**Comment by AyanSinhaMahapatra** [comment]
I'm getting a:
```
INFO: pip is looking at multiple versions of scancode-toolkit[packages,testing] to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install scancode-toolkit[packages,testing]==32.2.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    scancode-toolkit[packages,testing] 32.2.1 depends on go-inspector>0.3.0; platform_system == "Linux" and extra == "packages"
    The user requested (constraint) go-inspector==0.3.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```
On running `./configure --clean && ./configure --dev` not sure why this is not happening in the CI :thinking: 

Edit: seems to be a fluke, not happening anymore

**Comment by AyanSinhaMahapatra** [comment]
I think https://github.com/aboutcode-org/go-inspector/blob/main/src/go_inspector/binary.py#L482 should be in packagedcode here? Or do we need https://github.com/aboutcode-org/scancode-toolkit/pull/3761 to be merged before that?

**Comment by AyanSinhaMahapatra** [comment]
Thanks @pombredanne , merging!

**Pull request merged**